### PR TITLE
gcoap: small dependencies fixes

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

`gcoap` uses `random` but does not pull it indirectly, this fixes it. It's currently not shown because all possible network stacks pull in `random` for it.

### CODE REVIEW

File: sys/Makefile.dep [hunk 1]
```diff
@@ -648,9 +648,7 @@ ifneq (,$(filter gcoap,$(USEMODULE)))
   USEMODULE += ztimer_usec
   USEMODULE += event_callback
   USEMODULE += event_timeout_ztimer
-  ifneq (,$(filter openwsn%,$(USEMODULE)))
-    USEMODULE += openwsn_sock_udp
-  endif
```
Comment (aabadie) [inline sys/Makefile.dep:653]: This removal seems unrelated.

Comment (fjmolinas) [reply sys/Makefile.dep:653]: Its in the commit message [34eb656](https://github.com/RIOT-OS/RIOT/pull/17944/commits/34eb65642ccb53f58e0ffe5a8edabd34d4c30696)

**Comment by fjmolinas** [comment]
All green @benpicco

**Review by benpicco — APPROVED** [review]

**Comment by fjmolinas** [comment]
Thanks!
